atd-etl/cris_import/cris_import.py

```diff
@@ -225,45 +270,107 @@ def specify_extract_location():
+
+        cursor = pg.cursor()
 
-        zip_tmpdir = tempfile.mkdtemp()
-        rsync = None
-        try:
-            rsync = sysrsync.run(
-                verbose=True,
-                options=["-a"],
-                source_ssh=SFTP_ENDPOINT,
-                source="/home/txdot/*zip",
-                sync_source_contents=False,
-                destination=zip_tmpdir,
-                private_key=key_directory + "/id_ed25519",
-                strict_host_key_checking=False,
+        for obj in objects["Contents"]:
+            full_object_path = obj["Key"]
+            object_name = os.path.basename(full_object_path)
+
+            # Skip the "folder" object
+            if full_object_path == uploads_prefix:
+                continue
+
+            # Extract the S3 prefix from the full object path
+            object_path = os.path.dirname(full_object_path)
+
+            # Check if the object already exists in the database
+            cursor.execute(
+                "SELECT * FROM cris_import_log WHERE object_path = %s AND object_name = %s",
+                (object_path, object_name),
             )
-        except:
-            print("No files to copy..")
-            # we're really kinda out of work here, so we're going to bail
-            quit()
-        print("Rsync return code: " + str(rsync.returncode))
-        # check for a OS level return code of anything non-zero, which
-        # would indicate to us that the child proc we kicked off didn't
-        # complete successfully.
-        # see: https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html
-        if rsync.returncode != 0:
-            return False
-        print("Temp Directory: " + zip_tmpdir)
-        return zip_tmpdir
+            result = cursor.fetchone()
+
+            # If the object does not exist, insert a new record
+            if result is None:
+                cursor.execute(
+                    """
+                    INSERT INTO cris_import_log (object_path, object_name, first_seen)
+                    VALUES (%s, %s, %s)
+                    """,
+                    (object_path, object_name, datetime.datetime.utcnow()),
+                )
+
+        # Query all uploads where import_attempted = 0
+        cursor.execute("SELECT * FROM cris_import_log WHERE import_attempted is false")
+        uploads_to_import = cursor.fetchall()
+
+        # Check if there are no new files to process
+        if len(uploads_to_import) == 0:
+            raise Exception("No new files to process")
+
+        # Create a new temporary directory
+        import_dir = tempfile.mkdtemp()
+
+        # Download each upload into the new temporary directory
+        for upload in uploads_to_import:
+            upload_path = os.path.join(
+                upload[1], upload[2]
+            )  # Construct the full S3 path
+            upload_file_path = os.path.join(import_dir, upload[2])
+            s3.download_file(bucket, upload_path, upload_file_path)
```

Comment:
what would happen if we got into a situation where the DB thinks there should be a certain extract in the bucket, but for whatever reason that extract is not present? i assume `s3.download_file` would throw an error, right?

Reply:
Great idea! I'll have it check for successfully getting the zip down onto the local disk and to use `magic` to make sure it's a valid zip. https://github.com/cityofaustin/atd-vz-data/pull/1445/commits/01ec3ae7493da82518633e6238f3129a49299a36

Reply:
Additionally, the log table is not going to be used to track state -- it's just a log now -- and the existence of files in s3 is the only truth about what files are there or not, and that makes sense to me now. Thanks John 🙏

Reply:
The log is a real log, write and never read now. State is entirely held in the S3 filesystem.